src/web/pages.rs

```diff
@@ -1,29 +1,52 @@
 //! Routes to serve HTML pages
 
 use axum::{routing::get, Router};
+use axum_embed::ServeEmbed;
 use lazy_static::lazy_static;
+use rust_embed::Embed;
 use tera::Tera;
-use tower_http::services::ServeDir;
 
 use crate::web::WebState;
 
 mod index;
 
+#[derive(Embed, Clone)]
+#[folder = "src/web/static"]
+#[include = "*.css"]
+struct StaticAssets;
+
+#[derive(Embed)]
+#[folder = "src/web/templates"]
+#[include = "*.html"]
+struct Templates;
+
 pub fn compose(app_state: WebState) -> Router {
     Router::new()
         .route("/", get(index::index))
         .with_state(app_state)
-        .nest_service("/static", ServeDir::new("src/web/static"))
+        .nest_service("/static", ServeEmbed::<StaticAssets>::new())
 }
 
 lazy_static! {
     pub static ref TEMPLATES: Tera = {
-        match Tera::new("src/web/templates/**/*") {
-            Ok(t) => t,
-            Err(e) => {
-                println!("Template parsing error(s): {e}");
-                ::std::process::exit(1);
+        let mut tera = Tera::default();
+        for path in Templates::iter() {
+            let file = Templates::get(&path).unwrap();
+            let contents = match std::str::from_utf8(file.data.as_ref()) {
+                Ok(c) => c,
+                Err(e) => {
+                    println!("Template parsing error(s): {e}");
+                    ::std::process::exit(1);
```

Comment:
Sorry, missed this:

**suggestion**: Use `panic!`, will do essentially the same thing

Reply:
oh neat! I just switched that out.